In [1]:
import pandas as pd
import plotly.express as px

In [18]:
df = pd.read_csv("Compute_allocations2.csv")
# byte_level = df[df['Patch size'] == 1]
byte_level = df[(df['FLOPs'] == 's') & (df['Patch size'] == 2)]
# byte_level['Global transformer parameters'].sort_values(ascending=False)
fig = px.histogram(byte_level['Global model parameters'])

fig.show()

In [25]:
byte_level[(byte_level['Global model parameters'] <= 1.85e8) & (byte_level['Global model parameters'] >= 1.75e8) & (byte_level['Global model layers'] >= 9)].sort_values(by='Global model parameters', ascending=False)

,FLOPs,Patch size,Tokens,Encoder parameters,Global model parameters,Decoder parameters,Global model dimension,Encoder/Decoder dimension,Global model layers,Decoder layers
1823,s,2.0,9.760791e+09,4629053.44,1.816574e+08,18516213.76,1088,704,13,4.0
2106,s,2.0,1.035488e+10,3098787.84,1.816574e+08,15493939.20,1088,576,13,5.0
1598,s,2.0,9.915595e+09,5508956.16,1.816574e+08,16526868.48,1088,768,13,3.0
1600,s,2.0,9.445656e+09,6465372.16,1.816574e+08,19396116.48,1088,832,13,3.0
2434,s,2.0,9.922650e+09,3098787.84,1.816574e+08,18592727.04,1088,576,13,6.0
2432,s,2.0,1.055373e+10,2448424.96,1.816574e+08,14690549.76,1088,512,13,6.0
1821,s,2.0,1.029204e+10,3825664.00,1.816574e+08,15302656.00,1088,640,13,4.0
2108,s,2.0,9.772453e+09,3825664.00,1.816574e+08,19128320.00,1088,640,13,5.0
2207,s,2.0,9.851474e+09,3825664.00,1.810511e+08,19128320.00,896,640,19,5.0
2524,s,2.0,1.000413e+10,3098787.84,1.810511e+08,18592727.04,896,576,19,6.0


8e18 FLOPs
===

Patch size = 1.5

1131: 9e9 tokens, 163M parameters, 832 global model dimension, 512 encoder/decoder dimension, 20 global model layers, 6 decoder layers

628: 1.18e10 tokens, 119M parameters, 768 global model dimension, 576 encoder/decoder dimension, 17 global model layers, 4 decoder layers

400: 1.63e10 tokens, 82M parameters, 704 global model dimension, 576 encoder/decoder dimension, 14 global model layers, 3 decoder layers

321: 3.37e10 tokens, 39M parameters, 576 global model dimension, 384 encoder/decoder dimension, 10 global model layers,	3 decoder layers

Patch size = 2

2524: 1e10 tokens, 181M parameters,	896	global model dimension, 576	encoder/decoder dimension, 19 global model layers, 6 decoder layers

2218: 1.23e10 tokens, 139M parameters, 768 global model dimension, 576	encoder/decoder dimension, 20 global model layers, 5 decoder layers

1853: 1.72e10 tokens, 105M parameters, 768 global model dimension, 512 encoder/decoder dimension, 15 global model layers, 4 decoder layers

1531: 3.98e10 tokens, 43M parameters, 576 global model dimension, 384 encoder/decoder dimension, 11 global model layers, 3 decoder layers
